packages/node/src/typescript.ts

```diff
@@ -271,6 +271,17 @@ export function register(opts: Options = {}): Register {
       getCompilationSettings: () => config.options,
       getDefaultLibFileName: () => ts.getDefaultLibFilePath(config.options),
       getCustomTransformers: () => transformers,
+      resolveModuleNames: (moduleNames: string[], containingFile: string) => {
```

Comment:
I can't find a good way to write a test for what this fixes. A couple of times we've seen an inconsistent build error when tsconfig has `"strict": true` because the typescript program can't find `@types/express`. 

I was able to reproduce this using `pnpm` but our unit tests in this repo often use npm because the repo itself uses pnpm and the same error doesn't reproduce when used in this workspace.